### PR TITLE
perf(subscriptions): Add various metrics to subscriptions and incidents.

### PR DESCRIPTION
**Pull request by wedamija**
We want to build a dashboard to help monitor subscriptions and incidents, adding some more details
stats and timings to help facilitate.

### CODE REVIEW

File: src/sentry/incidents/subscription_processor.py [hunk 1]
```diff
@@ -142,13 +142,15 @@ def process_update(self, subscription_update):
...
             ) and not self.check_trigger_status(trigger, TriggerStatus.ACTIVE):
+                metrics.incr("incidents.alert_rules.hit_trigger_threshold")
                 with transaction.atomic():
                     self.trigger_alert_threshold(trigger)
             elif (
                 trigger.resolve_threshold is not None
                 and resolve_operator(aggregation_value, trigger.resolve_threshold)
                 and self.check_trigger_status(trigger, TriggerStatus.ACTIVE)
             ):
+                metrics.incr("incidents.alert_rules.hit_resolve_threshold")
```
Comment (markstory) [inline src/sentry/incidents/subscription_processor.py]: Would this and the trigger metric be better as tags? If the metric name was `incidents.alert_rules.threshold` with hit/resolved as tags you could graph the two states in datadog a bit more easily.

File: src/sentry/incidents/subscription_processor.py [hunk 2]
```diff
@@ -173,6 +175,7 @@ def trigger_alert_threshold(self, trigger):
         """
         self.trigger_alert_counts[trigger.id] += 1
         if self.trigger_alert_counts[trigger.id] >= self.alert_rule.threshold_period:
+            metrics.incr("incidents.alert_rules.fired_trigger")
```
Comment (markstory) [inline src/sentry/incidents/subscription_processor.py]: Same with this metric, fired/resolved could be tags.

**Review by markstory — APPROVED** [review]
Looks good to me. The tagging comments are just ideas that may or may not help you build dashboards/alerts more easily.

File: src/sentry/incidents/subscription_processor.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/incidents/subscription_processor.py]: Ahh I see. I'm still used to the old graphite way of doing things, this makes sense though.

**Pull request merged**